### PR TITLE
MSLRES-5299: Update SDK with MMS MO feature support

### PR DESCRIPTION
**Pull request by gkvasnsinch**

### CODE REVIEW

**Review by GSvensk — CHANGES_REQUESTED** [review]

File: src/main/java/com/sinch/xms/api/MoSms.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
  * MoTextSms textual} or a {@link MoBinarySms binary} message payload.
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
-@JsonSubTypes({@Type(MoTextSms.class), @Type(MoBinarySms.class)})
+@JsonSubTypes({@Type(MoTextSms.class), @Type(MoBinarySms.class), @Type(MoMms.class)})
```
[the remark this answers is not in the document]
Comment (gkvasnsinch) [reply src/main/java/com/sinch/xms/api/MoSms.java:34]: Though it is not really fits there, as mms under sms 🤷

File: src/main/java/com/sinch/xms/api/MoMmsMedia.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import javax.annotation.Nonnull;
+import org.immutables.value.Value;
+
+/** A mobile originated MMS message. */
+@Value.Immutable
+@ValueStylePackage
+@JsonDeserialize(as = MoMmsMediaImpl.class)
+@JsonTypeName("mo_media")
+public abstract class MoMmsMedia {
```
[the remark this answers is not in the document]
Comment (gkvasnsinch) [reply src/main/java/com/sinch/xms/api/MoMmsMedia.java:32]: Which information do you mean specifically?

File: src/main/java/com/sinch/xms/api/MoSms.java [hunk 3]
```diff
@@ -28,10 +28,10 @@
 
 /**
  * Base class for mobile originated messages. A mobile originated message can have either a {@link
- * MoTextSms textual} or a {@link MoBinarySms binary} message payload.
+ * MoTextSms textual} or a {@link MoBinarySms binary} or a {@link MoMms} message payload.
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
-@JsonSubTypes({@Type(MoTextSms.class), @Type(MoBinarySms.class)})
+@JsonSubTypes({@Type(MoTextSms.class), @Type(MoBinarySms.class), @Type(MoMms.class)})
```
Comment (thomasf147) [inline src/main/java/com/sinch/xms/api/MoSms.java:34]: A philosophical question is if MoMms is a MoSms (the inheritance)

Comment (gkvasnsinch) [reply src/main/java/com/sinch/xms/api/MoSms.java:34]: It is not, but renaming MoSms => MoMessage might be more problematic I guess 🤷

Comment (thomasf147) [reply src/main/java/com/sinch/xms/api/MoSms.java:34]: Yes, is simply Mo.class. Indeed it would break BC. I think your current solution might not break BC

Comment (GSvensk) [reply src/main/java/com/sinch/xms/api/MoSms.java:34]: I prefer not to break BC

Comment (thomasf147) [reply src/main/java/com/sinch/xms/api/MoSms.java:34]: Came up with a funnier name though MoXms

**Review by thomasf147 — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -3,7 +3,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.sinch</groupId>
   <artifactId>sdk-sms</artifactId>
-  <version>2.0.2-SNAPSHOT</version>
+  <version>2.0.4-SNAPSHOT</version>
```
Comment (thomasf147) [inline pom.xml:6]:
```suggestion
  <version>2.0.3-SNAPSHOT</version>
```

**Pull request merged**